Python/stackrefs.c

```diff
@@ -197,4 +197,27 @@ _PyStackRef_CLOSE_SPECIALIZED(_PyStackRef ref, destructor destruct, const char *
     _Py_DECREF_SPECIALIZED(obj, destruct);
 }
 
+_PyStackRef PyStackRef_TagInt(intptr_t i)
+{
+    return (_PyStackRef){ .index = (i << 1) + 1 };
```

Comment:
Wait why << 1 only? Isn't it << 2 according to your tagging scheme?

Also does this handle signed negatives? Doesn't feel like it to me.

Reply:
This is with `Py_STACKREF_DEBUG`. It works entirely differently to the normal tagging scheme.

Reply:
:facepalm: ok I see, that explains it.